runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -277,26 +331,15 @@ MM_IndexableObjectAllocationModel::layoutDiscontiguousArraylet(MM_EnvironmentBas
 		case GC_ArrayletObjectModel::Discontiguous:
 			indexableObjectModel->AssertArrayletIsDiscontiguous(spine);
 			Assert_MM_true(arrayoidIndex == _numberOfArraylets);
-#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
-			if (indexableObjectModel->isDoubleMappingEnabled()) {
-				/**
-				 * There are some special cases where double mapping an arraylet is
-				 * not necessary; isArrayletDataDiscontiguous() details those cases.
-				 */
-				if (indexableObjectModel->isArrayletDataDiscontiguous(spine)) {
-					doubleMapArraylets(env, (J9Object *)spine, NULL);
```

Comment:
check why we did not simply the code block above where we call allocateArrayletLeaf, since we now do it later in reserveLeavesForContiguousArraylet

Reply:
double map as an alternative of sparse heap allocation for InlineContiguous layout case, putting doubleMapArraylets() in  reserveLeavesForContiguousArraylet() for double map would match allocateSparseFreeEntryAndMapToHeapObject() for off-heapin getSparseAddressAndDecommitLeaves().